app/Jobs/EntityWebhookJob.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+namespace App\Jobs;
+
+use App\Http\Resources\EntityResource;
+use App\Models\Entity;
+use App\Models\Webhook;
+use App\Models\Campaign;
+use App\User;
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Foundation\Bus\Dispatchable;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Queue\SerializesModels;
+use Illuminate\Support\Facades\Log;
+use Illuminate\Support\Facades\Http;
+use Illuminate\Support\Str;
+use App\Facades\Avatar;
+
+class EntityWebhookJob implements ShouldQueue
+{
+    use Dispatchable;
+    use InteractsWithQueue;
+    use Queueable;
+    use SerializesModels;
+
+    /**
+     * @var int
+     */
+    public $campaignId;
+
+    /**
+     * @var int
+     */
+    public $action;
+
+    /**
+     * @var string
+     */
+    public $username;
+
+    /**
+     * @var Entity
+     */
+    public $entity;
+
+    /**
+     * The number of times the job may be attempted.
+     *
+     * @var int
+     */
+    public $tries = 1;
+
+    /**
+     * Create a new job instance.
+     *
+     * @return void
+     */
+    public function __construct(Entity $entity, Campaign $campaign, User $user, int $action)
+    {
+        // Can't save the entity directly into the job because of the child() function not returning a
+        // string? Maybe something to do with the to array part of the queue.
+        $this->campaignId = $campaign->id;
+        $this->username = $user->name;
+        $this->action = $action;
+        $this->entity = $entity;
+    }
+
+    /**
+     * Execute the job.
+     *
+     * @return void
+     */
+    public function handle()
+    {
+        Log::info('EntityWebhookJob for entity #' . $this->entity->id);
+
+        $webhooks = Webhook::where('campaign_id', $this->campaignId)->where('action', $this->action)->where('status', 1)->get();
+
+        foreach ($webhooks as $webhook) {
+            if ($webhook->tags()->count() > 0) {
+                $tags = $webhook->tags()->pluck('tags.id')->all();
+
+                if (!$this->entity->tags()->whereIn('tags.id', $tags)->first()) {
```

Comment:
Is this setup so that entities need _all the tags_ or _at least one of the tags_? Because whichever version it is, we need to warn users about it or they'll get confused and spam the discord with the same question.

Reply:
At least one of the tags.